pkg/api/folder.go

```diff
@@ -875,22 +870,29 @@ func (fk8s *folderK8sHandler) newToFolderDto(c *contextmodel.ReqContext, item un
 		}
 		metrics.MFolderIDsAPICount.WithLabelValues(metrics.NewToFolderDTO).Inc()
 
-		fDTO.CanSave = canSave
-		fDTO.CanEdit = canEdit
-		fDTO.CanAdmin = canAdmin
-		fDTO.CanDelete = canDelete
-		fDTO.CreatedBy = creator
-		fDTO.UpdatedBy = updater
-		fDTO.AccessControl = acMetadata
-		fDTO.OrgID = f.OrgID
-		// #TODO version doesn't seem to be used--confirm or set it properly
-		fDTO.Version = 1
-
-		return *fDTO, nil
+		return dtos.Folder{
+			ID:        fold.ID, // nolint:staticcheck
+			UID:       fold.UID,
+			Title:     fold.Title,
+			URL:       fold.URL,
+			HasACL:    fold.HasACL,
+			CanSave:   canSave,
+			CanEdit:   canEdit,
+			CanAdmin:  canAdmin,
+			CanDelete: canDelete,
+			CreatedBy: creator,
+			Created:   fold.Created,
+			UpdatedBy: updater,
+			Updated:   fold.Updated,
+			// #TODO version doesn't seem to be used--confirm or set it properly
+			Version:       1,
+			AccessControl: acMetadata,
+			ParentUID:     fold.ParentUID,
+		}, nil
 	}
 
 	// no need to check view permission for the starting folder since it's already checked by the callers
-	folderDTO, err := toDTO(f, false)
+	folderDTO, err := toDTO(f, createdBy, false)
```

Comment:
Suggestion: Perhaps we can add two consts for what false and true would mean in `toDTO`. So we end up with something like 

```golang
folderDTO, err := toDTO(f, createdBy, dontCheckCanView)
// or
folderDTO, err := toDTO(f, createdBy, checkCanView)
```

Reply:
I like it. Thanks for the review! Applied changes for both suggestions here: [5f46dae](https://github.com/grafana/grafana/pull/96439/commits/5f46daeb912dd57fb3e0cbf50be610a99ab9738e)